day_09.md

```diff
@@ -0,0 +1,265 @@
+
+---
+
+### Callback
+
+> A callback function is a function passed into another function as an argument, which is then invoked inside the outer function to complete some kind of routine or action.
+>
+> Here is a quick example:
+>
+>```javascript
+> function greeting(name) {
+>  alert('Hello ' + name);
+>}
+>
+>function processUserInput(callback) {
+>  var name = prompt('Please enter your name.');
+>  callback(name);
+>}
+>
+>processUserInput(greeting);
+>```
+>
+> The above example is a synchronous callback, as it is executed immediately.
+>
+> Note, however, that callbacks are often used to continue code execution after an asynchronous operation has completed — these are called asynchronous callbacks. A good example is the callback functions executed inside a .then() block chained onto the end of a promise after that promise fulfills or rejects. This structure is used in many modern web APIs, such as fetch().
+>
+> Source: [MDN - Callback Function](https://developer.mozilla.org/en-US/docs/Glossary/Callback_function)
+
+Let's take a look at a [general definition of callback on Wikipedia](https://en.wikipedia.org/wiki/Callback_(computer_programming))
+
+Here an simplified example of callback you might have used but now you'll know better what id does.
+
+```javascript
+setTimeout( () => console.log(`I'm running late...er`), 1000 );
+```
+
+Can you describe, in your words like a bedtime story, what's happening with that code? Need a help with that? Let's move on and take a final step into this with [YDKJS - Async & Performance - Callbacks](https://github.com/getify/You-Dont-Know-JS/blob/1st-ed/async%20%26%20performance/ch2.md) | by Kyle Simpson
+
+---
+
+### Promises
+
+Let's imagine we need to perform a task, you don't know how much it's gonna take but we need both make sure to do something whilst it's done AND in the meantime do something else AND have a complete control over the outcomes to prevent weir things happening.
+
+If we do that with callbacks we'll end up in hell, actually there's a name for that, [callback hell](https://en.wiktionary.org/wiki/callback_hell), and there's even a [website](http://callbackhell.com/) with that name!
+
+Let's make and example.
+
+```javascript
+
+setTimeout( () => {
+  console.log(`A`);
+  setTimeout( () => {
+    console.log(`B`);
+    setTimeout( () => {
+      console.log(`C`);
+      setTimeout( () => {
+        console.log(`D`);
+      }, 1000 );
+    }, 100 );
+  }, 0 );
+}, 5000 );
+
+```
+
+After a few moments thinking about it you might say "that's easy", I won't gent into the "using `console`" implications, but let's imagine that you need `B`to happen only if `A`  is successful, AND `C`AND `D` should happen either way. Now our code starts looking not only hard to read but also hard to maintain, which is the doorway to a tsunami of bugs swarming to you.
+
+So, what's a promise and why it might help here?
```

Comment:
_why it might help here?_ sounds weird because it's a question. May be you can ask _How can/could it help here?_

Reply:
I'll keep the why but in a separated sentence. In this context, I'm more interested in the "why" instead of  the "how"